src/printer.js

```diff
@@ -2053,6 +2053,20 @@ function printJSXElement(path, options, print) {
         } else {
           children.push(print(childPath));
         }
+
+        if (
+          child.closingElement &&
+            child.closingElement.type === 'JSXClosingElement'
+        ) {
+          const nextChild = n.children[index + 1];
+
+          if (
+            typeof nextChild === 'undefined' ||
+              nextChild.type !== 'JSXText'
+          ) {
+            children.push(hardline);
```

Comment:
My hunch is that `softline` should be sufficient; some people may want code like

```jsx
<p>hello <span>there</span></p>
```

and hard-breaking in that case may not always result in prettier code.